### PR TITLE
Align to 9.4.0

### PR DESCRIPTION
**Pull request by d4r1091**
## Context

In light of the recent release, we want to align the Firefox Upgrade branch to the latest 9.4.0 version.

## Approach

- Cherry-picked commits
- Went thru the commits merged in `main`
- Analyzed discrepancies given the new project layout and dependencies (ShareTo + Clients) in the Firefox-upgrade branch
- Brought changes into the respective new files
- Executed also a "Build for Testing" to double check dependencies in the new testing layout

### Checklist

- [x] I performed some relevant testing on a real device and/or simulator

### CODE REVIEW

File: Client/Frontend/Browser/SearchViewController.swift [hunk 1]
```diff
@@ -583,6 +583,9 @@ class SearchViewController: SiteTableViewController,
                   let url = defaultEngine.searchURLForQuery(suggestion)
             else { return }
 
+            // Ecosia: increment search count when bing distribution
+            BingDistributionExperiment.incrementCounterIfTestVariant()
```
Comment (lucaschifino) [inline Client/Frontend/Browser/SearchViewController.swift:587]: We had two places where the increment should happen, one was this one and then the other was `BrowserViewController+URLBarDelegate` as shown in [the PR](https://github.com/ecosia/ios-browser/pull/650/files).

I didn't see this second part here though, did this change with the upgrade somehow?

File: Client/Ecosia/Settings/EcosiaDebugSettings.swift
Comment (lucaschifino) [inline Client/Ecosia/Settings/EcosiaDebugSettings.swift:1]: I'm also missing the changes to `AppSettingsTableViewController` where the search settings were not added, is this something that changed as well?

**Review by lucaschifino — COMMENTED** [review]
Nice 👏 

Added two comments about things that are potentially still missing, let me know what you think :)

File: Client/Ecosia/Settings/EcosiaDebugSettings.swift
Comment (d4r1091) [reply Client/Ecosia/Settings/EcosiaDebugSettings.swift:1]: Thanks for the double check 👏 .
The `AppSettingsTableViewController` has a separate file `+Ecosia` that contains the changes to apply.
Done it there now 💪

File: Client/Frontend/Browser/SearchViewController.swift (hunk 1 above)
Comment (d4r1091) [reply Client/Frontend/Browser/SearchViewController.swift:587]: Awesome! The cherry pick didn't show this as a conflict for some reason but I added it after double checking your comment.

**Review by d4r1091 — COMMENTED** [review]
Done with the suggestions @lucaschifino 💪 
Thanks for the review!

**Review by lucaschifino — APPROVED** [review]
LGTM 🟢

**Pull request merged**